bundle/src/main/java/com/adobe/acs/commons/wcm/impl/PropertyMergePostProcessor.java

```diff
@@ -21,183 +21,257 @@
-            }
-        }
+                    });
+        });
 
         // Convert the Mappings into PropertyMerge objects
+        return mapping.entrySet().stream().map(
+                entry -> new PropertyMerge(
+                        entry.getKey(),
+                        entry.getValue(),
+                        areDuplicatesAllowed(requestParameters, entry.getKey()),
+                        getFieldTypeHint(requestParameters, entry.getKey())
+                ))
+                .collect(Collectors.toList());
+    }
 
-        final List<PropertyMerge> propertyMerges = new ArrayList<PropertyMerge>();
-
-        for (final Map.Entry<String, List<String>> entry : mapping.entrySet()) {
-            final String destination = entry.getKey();
-            final List<String> sources = entry.getValue();
+    private void trackMergeParameters(final HashMap<String, Set<String>> mapping, final String source, String destination) {
+        mapping.merge(destination, new HashSet<String>(), (a, b) -> a).add(source);
+    }
 
-            RequestParameter allowDuplicatesParam = requestParameterMap.getValue(IGNORE_PREFIX + destination
-                    + ALLOW_DUPLICATES_SUFFIX);
+    private void trackAssetMergeParameters(final RequestParameterMap requestParameters, final String source, String destination, final HashMap<String, Set<String>> mapping) {
+        requestParameters.keySet().stream()
+                .map(String::valueOf)
+                .filter((paramName) -> (paramName.endsWith("/" + source)))
+                .forEach(adjustedSource -> {
+                    String adjustedDest = alignDestinationPath(adjustedSource, destination);
+                    trackMergeParameters(mapping, adjustedSource, adjustedDest);
+                });
+    }
 
-            final boolean allowDuplicates = allowDuplicatesParam != null && Boolean.valueOf(allowDuplicatesParam.getString());
+    private void trackAllTagsMergeParameters(RequestParameterMap requestParameters, String destination, HashMap<String, Set<String>> mapping) {
+        requestParameters.forEach((source, value) -> {
+            if (hasTags(value)) {
+                String newDestination = alignDestinationPath(source, destination);
+                trackMergeParameters(mapping, source, newDestination);
+            }
+        });
+    }
 
-            RequestParameter typeHintParam = requestParameterMap.getValue(IGNORE_PREFIX + destination
-                    + TYPE_HINT_SUFFIX);
+    protected static boolean hasTags(RequestParameter[] params) {
+        if (params == null) {
+            return false;
+        } else {
+            return Stream.of(params).allMatch(param -> looksLikeTag(param.getString()));
+        }
+    }
 
-            final String typeHint =
-                    typeHintParam != null ? typeHintParam.getString() : String.class.getSimpleName();
+    protected static boolean looksLikeTag(String value) {
+        return VALID_TAG.asPredicate().test(value);
```

Comment:
I wonder if this should be a bit more stringent and use `tagManager.resolve(..)` to *make sure* its a tag, vs that it just "looks like" a tag.

Reply:
I thought about that -- but I also thought about how slow TagManager is, too.

Reply:
The TagManager.resolve(..) code does not look slow. Why do you think its slow, and prohibitively slow for an activity that is bounded by human interaction?

Reply:
I have my reservations about TagManager in general, but more to the point this has to survive bulk editing too, so any additional work can become multiplicative.  If we use TagManager, we should also consider a small memory-limited cache to avoid looking up the same tags over and over.

Reply:
@badvision have you looked at the resolve code? it seems very simple (string checks).. or am i missing something? I dont think that this check will make or break anything in the context of writes; i'd be far more concerned around ~everything else (oak write overheads, any property indexes effected, WF's via launchers getting triggered). 

Do you have experience w `tm.resolve(..)` specifically where its prohibitively slow?  You seem very concerned about :) .. i feel like perhaps im missing something.

Reply:
If you've moved or renamed tags, it's not that simple.  The whole implementation is a mess I would rather avoid.  But if this is what you want to accept the PR, then consider it added.  I have reservations about it, but pre-mature optimizations are the root of all evil so I'll shelve the caching idea for now.